addons/godot-xr-tools/functions/movement_grapple.gd

```diff
@@ -134,14 +134,27 @@ func _process(_delta: float):
 		_line.visible = false
 
 	# Update grapple target
-	if enabled and !is_active and _is_raycast_valid():
-		_grapple_target.global_transform.origin  = _grapple_raycast.get_collision_point()
+	if enabled and !is_active and is_instance_valid(hook_object):
```

Comment:
Wouldn't it make more sense to move this into _physics_process in its entirety?

Reply:
Yeah, I was trying to keep rendering and physics updates separate; but if/when we do support running physics in a separate thread this will be the least of our concerns. I've merged them together into _physics_process.